src/JwtCredentialsManager.ts

```diff
@@ -19,7 +19,7 @@ export default abstract class JwtCredentialsManager<
 
   token: string | null | undefined;
 
-  credentials: TCredentials | null | undefined;
+  credentials: Promise<TCredentials | null | undefined>;
```

Comment:
**nitpick:** just to make things a bit more clear

```suggestion
  credentialsPromise: Promise<TCredentials | null | undefined>;
```

Reply:
you probably want to make the typing here include `null` and initialize this to `null`, or something to that effect, too

Reply:
or maybe:

```suggestion
  credentials: Promise<TCredentials | null | undefined> | TCredentials | null | undefined;
```

actually i'm not sure `undefined` should even be allowed here.